packages/gatsby-theme-newrelic/src/components/GlobalHeader.js

```diff
@@ -52,6 +68,10 @@ const GlobalHeader = ({ editUrl, className, search }) => {
     siteMetadata: { repository },
   } = site;
 
+  window.history.pushState = track(window.history.pushState, () => {
```

Comment:
Rather than using the built-in `pushState` call, can we instead use Gatsby's `navigate` function? I'm not sure if Gatsby tracks state internally, so it would be best to try and use the Gatsby APIs as much as possible. https://www.gatsbyjs.org/docs/gatsby-link/#how-to-use-the-navigate-helper-function

Reply:
I stand corrected. They recommend using the `navigate` function from reach router directly when routing query params: https://www.gatsbyjs.org/docs/gatsby-link/#recommendations-for-programmatic-in-app-navigation

Reply:
Another note here, I see you're rewriting `window.history.pushState` inside of the component, which means this is going to be defined across the entire app as soon as this component is used. If this component ever unmounts (I know this is the global header, but the same principle applies), you might have a memory leak, or at least you'll confuse React because you'll be calling `setIsQueryChanged` even after the component no longer exists. ~Typically~ Always avoid overwriting globals, especially for APIs that other parts of the app depend on, and especially if it adds behavior specific to a single component. It gets hard to untangle later if you're not careful about cleaning this up.

As mentioned, the `navigate` function should help you here.